lib/src/text_parser.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'charcode.dart';
+
+/// A parser to parse a segment of source text.
+class TextParser {
```

Comment:
The purpose  of this `TextParser` is to parse a text segment separately. It will be reused later when I am working on fixing the link and link reference definitions issues.